test/recipes/02-test_errstr.t

```diff
@@ -40,79 +40,52 @@ plan skip_all => 'This is unsupported on MSYS/MinGW or MSWin32'
-    # version, we resort to discovering them directly in the Errno package
-    # symbol table.
-    my @error_skiplist = qw(
-        ENETDOWN
-        ENETUNREACH
-        ENETRESET
-        ECONNABORTED
-        EISCONN
-        ENOTCONN
-        ESHUTDOWN
-        ETOOMANYREFS
-        ETIMEDOUT
-        EHOSTDOWN
-        EHOSTUNREACH
-        EALREADY
-        EINPROGRESS
-        ESTALE
-        EUCLEAN
-        ENOTNAM
-        ENAVAIL
-        ENOMEDIUM
-        ENOKEY
-    );
-    @posix_errors =
-        grep {
-            my $x = $_;
-            ! grep {
-                exists $Errno::{$_} && $x == $Errno::{$_}
-            } @error_skiplist
-        } @posix_errors;
-}
+# We use Errno::EXPORT_OK as a list of known errno values on the current
+# system.  libcrypto's ERR should either use the same string as perl, or if
+# it was outside the range that ERR looks at, ERR gives the reason string
+# "reason(nnn)", where nnn is the errno number.
 
-plan tests => scalar @posix_errors
+plan tests => scalar @Errno::EXPORT_OK
     +1                          # Checking that error 128 gives 'reason(128)'
     +1                          # Checking that error 0 gives the library name
     ;
 
-foreach my $errname (@posix_errors) {
+foreach my $errname (@Errno::EXPORT_OK) {
+    # The error names are perl constants, which are implemented as functions
+    # returning the numeric value of that name.
     my $errnum = "Errno::$errname"->();
 
- SKIP: {
-        skip "Error $errname ($errnum) isn't within our range", 1
-            if $errnum > NUM_SYS_STR_REASONS;
-
-        my $perr = eval {
-            # Set $! to the error number...
-            local $! = $errnum;
-            # ... and $! will give you the error string back
-            $!
-        };
-
-        # We know that the system reasons are in OpenSSL error library 2
-        my @oerr = run(app([ qw(openssl errstr), sprintf("2%06x", $errnum) ]),
-                       capture => 1);
-        $oerr[0] =~ s|\R$||;
-        @oerr = split_error($oerr[0]);
-        ok($oerr[3] eq $perr, "($errnum) '$oerr[3]' == '$perr'");
+  SKIP: {
+      # ERR only handles errnos that fit in 24 bits.  Numbers beyond that are
+      # trunkated and give false results.
+      skip "$errnum is larger than a 24 bit value, not supported", 1
+          if ($errnum & ~0xFFFFFF) != 0;
+
+      my $perr = eval {
+          # Set $! to the error number...
+          local $! = $errnum;
+          # ... and $! will give you the error string back
+          $!
+      };
+
+      # We know that the system reasons are in OpenSSL error library 2
+      my @oerr = run(app([ qw(openssl errstr), sprintf("2%06x", $errnum) ]),
+                     capture => 1);
+      $oerr[0] =~ s|\R$||;
+      @oerr = split_error($oerr[0]);
+      # Optional, if libcrypto didn't capture the text
+      my $rerr = "reason($errnum)";
+      ok($oerr[3] eq $perr || $oerr[3] eq $rerr,
```

Comment:
This would be much easier to read if it were a single if statement with two different ok calls.

Reply:
It's a trade-off.  If you want to quickly check that you've calculated the plan right, it's easier to see that there's only one `ok` per code and not two if there's *actually* only one `ok` there.

Reply:
```
if (something) { ok ... }
else { ok ... }
```
seems pretty obvious to me.  The proposed code is far from obvious to me.

Reply:
At least three people think that that `ok(P || Q, P? ... : ...)` call is needlessly hard to read.
On the other hand, well, it's "just" part of a test script.